From eeff1599dea13499d2013bcbe28ab5ce774962b4 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@suse.de>
Date: Fri, 7 Nov 2014 00:01:48 +0000
Subject: [PATCH] x86, cpu: Detect more TLB configuration -xen

References: TLB Performance
Patch-mainline: No, only required for xen

Copy across come 1G TLB information. It is severely doubtful that Xen even
can use this informatioon but it's convenient to have consistent reporting
of TLB information.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 arch/x86/include/mach-xen/asm/processor.h | 1 +
 arch/x86/kernel/cpu/common-xen.c          | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/x86/include/mach-xen/asm/processor.h b/arch/x86/include/mach-xen/asm/processor.h
index 99b4862..efcedbf 100644
--- a/arch/x86/include/mach-xen/asm/processor.h
+++ b/arch/x86/include/mach-xen/asm/processor.h
@@ -74,6 +74,7 @@ extern u16 __read_mostly tlb_lli_4m[NR_INFO];
 extern u16 __read_mostly tlb_lld_4k[NR_INFO];
 extern u16 __read_mostly tlb_lld_2m[NR_INFO];
 extern u16 __read_mostly tlb_lld_4m[NR_INFO];
+extern u16 __read_mostly tlb_lld_1g[NR_INFO];
 extern s8  __read_mostly tlb_flushall_shift;
 
 /*
diff --git a/arch/x86/kernel/cpu/common-xen.c b/arch/x86/kernel/cpu/common-xen.c
index d932aff..474c34b 100644
--- a/arch/x86/kernel/cpu/common-xen.c
+++ b/arch/x86/kernel/cpu/common-xen.c
@@ -525,6 +525,7 @@ u16 __read_mostly tlb_lli_4m[NR_INFO];
 u16 __read_mostly tlb_lld_4k[NR_INFO];
 u16 __read_mostly tlb_lld_2m[NR_INFO];
 u16 __read_mostly tlb_lld_4m[NR_INFO];
+u16 __read_mostly tlb_lld_1g[NR_INFO];
 
 /*
  * tlb_flushall_shift shows the balance point in replacing cr3 write
